### DISCUSSION TITLE
Version 0.6.1 (final)

### DISCUSSION THREAD

**Discussion by DSheirer** [DSheirer/sdrtrunk#2100] [Announcements]
## Features:
* Tuner minimum and maximum frequency now user adjustable so you can lock a tuner to a specific frequency range
* SDRPlay RSP1B & RSPdxR2 tuners and API 3.15/3.16 support
* P25 Phase 1 & 2 expanded message decoding, enhanced traffic channel management, talker alias/gps and ISSI/Roam fully qualified Identifiers with alias support
* DMR significant improvements to processing and talker alias/gps support
* Enhanced map display
* Radio Reference API version 18
* Fix for some critical issues causing out-of-memory, application crashes, and channels stuck in TEARDOWN
* Automatic threading dead-lock detection, notification and report creation
* Create diagnostic reports (File >> Reports) for reporting out-of-memory and app crash errors
* Windows aarch64 support
* Latest Java OpenJDK 23

## Requirements:
* Windows x86/aarch64 (64-bit)
* Linux x86/aarch64 (64-bit)
* OSX x86/aarch64 (64-bit)

**Note**: Linux for arm-32 is NOT supported

## Optional Requirements
**JMBE version 1.0.9+** is required for decoding APCO25 Phase 1/2 and DMR digital audio on Linux and OSX aarch64 and this library can be compiled within the sdrtrunk application.

Legacy instructions for compiling the JMBE library are located at: **https://github.com/DSheirer/sdrtrunk/wiki/JMBE**

## User Manual
* [Wiki](https://github.com/DSheirer/sdrtrunk/wiki)
* [User Manual](https://github.com/DSheirer/sdrtrunk/wiki/User-Manual)

## Support
* [Support](https://github.com/DSheirer/sdrtrunk/wiki/Support)

## What's Changed
* 1772 Patch Group Streaming Preference (Patch Group vs Individual Talkgroups) by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1773
* #1783 P25P2 L3Harris Talker Alias Index Out Of Bounds Exception by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1784
* #1783 L3H P25P2 Talker Alias Guard Against Negative Length Aliases by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1786
* #1787 Enhance DMR CSBK CRC Calculations - Auto-Detect Alternate CRC Mask Values by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1793
* #1791 Resolves OpenMHz Broadcaster Auto-Reconnection Issue by @tadscottsmith in https://github.com/DSheirer/sdrtrunk/pull/1791
* P25-P2 Encryption Synchronization ReSync by @crsawyer in https://github.com/DSheirer/sdrtrunk/pull/1792
* #1787 DMR Decoder Enhancements for RAS Enabled Systems by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1794
* #1787 DMR CapMax Group Voice Channel Update by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1796
* #1787 DMR Cap+ Rest Allocation & Channel Rotation Issue by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1798
* #1799 DMR Cap+ & CapMax now track events correctly.  Updated Cap+ eve… by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1801
* #1805 DMR details tab enhancements and opcode cleanup. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1806
* Inherit P25 Trunked Preferred Tuner To Traffic Channels by @Nokoa in https://github.com/DSheirer/sdrtrunk/pull/1808
* Use TLSv1.3 when connecting to OSM Tileserver by @poggs in https://github.com/DSheirer/sdrtrunk/pull/1813
* Fix dB power meter heading by @poggs in https://github.com/DSheirer/sdrtrunk/pull/1814
* #1812 SDRPlay  API version 3.14 & RSP1B by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1817
* #1819 SDRPlay RSP1B Tuner Detection by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1820
* #1821 SDRPlay IF Gain Adjustment by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1822
* #1824 Diagnostic Report Generation by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1825
* #1827 UI Thread Deadlock From USB Hotplug Support Callback by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1828
* #1802 Channels Stuck in TEARDOWN & Application Crash From Out Of Memory  by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1829
* #1830 Health and Diagnostic Monitoring by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1831
* #1826 SIMD-512 Systems Fail When Decoding MPT1327 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1832
* #1810 P25 Voice Frame List Concurrent Modification Error by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1833
* #1834 CapMax Adv Mode Call Events With NO FREQUENCY by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1835
* #1837 Channel Stream Thread Naming by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1838
* #1839 DMR Tone Identifiers Don't Clear At Call End by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1840
* #1841 DMR Call Events Include Channel & Frequency by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1842
* #1843 Map Entity Plotter Concurrent Modification Exception by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1844
* #1845 Tuner Don't Change Frequency Unless Have To by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1846
* #1847 Spectral Display of Multi-Frequency Channel Configurations by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1848
* #172 Tuner Supports Adjusting Min/Max Frequencies by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1850
* #1852 Spectral Display Channel Banner Coloration by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1853
* #1854 Tuner Min Frequency - Allow 1 Hertz Minimum by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1855
* #1856 DMR Channel - Threading Deadlock by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1857
* #1304 P25 TDMA CC Support & Phase 1/2 decoder enhancements. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1898
* #1899 Radio Reference API V18 & Update Site Editor To Show TDMA CC Sites by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1900
* #1876 Unifies Reentrant Lock Between Tuners and Frequency Controller. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1901
* #1903 P25 Phase 2 Audio Segment Chopping by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1904
* #1905 P25 Phase 1 Encrypted Call Status by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1906
* #1907 Hytera GPS & XPT Neighbor & Talker Aliases by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1908
* #1849 Map Enhancements by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1920
* #1924 GPS Positions Listed as User Identifiers In Map Panel by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1925
* #1916 SDRPlay API 3.15 and RSPdxR2 support. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1921
* #1926 Toolkit not initialized error when running headless during APCO… by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1928
* #1931 P25P2 L3Harris Patch Group - Incorrect Radio/Talkgroup Members by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1932
* #1934 P25P1 Conventional Fallback LC Message by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1935
* #1781 Radio Reference API Login Enhanced Logging by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1943
* #1944 SDRPlay API version 3.16 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1945
* 1860 LibUsb Transfer Buffer Error Handling Enhancements by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1950
* #1952 P25 Traffic Channel Manager Aggregating Multiple Call Events Into One by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1953
* #1912 P25 Patch Group Details Missing In Now Playing and Streaming/Recording Metadata by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1954
* #1955 Java JDK 22 and Gradle 8.10 and Version to beta2 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1958
* #1960 Application Root Directory  Not Found On Fresh Install by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1961
* #1964 SDRPlay API Native Library on MacOS by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1965
* #1966 P25 Aliases Not Correctly Matching Against Fully Qualified Radio & Talkgroup Values by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1967
* #1948 Audio Streaming Startups On Separate Thread by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1968
* #1974 P25 Unassigned Fully Qualified Talkgroup/Radio Value Handling by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1975
* #1971 P25 AMBTC Status Query - Error Parsing Source Address by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1976
* #1977 P25 FQSUID Improvements by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1978
* #1939 Duplicate Call Detector Enhancements & Unit Tests by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1980
* Improvements to icecast failure handling by @doug-hoffman in https://github.com/DSheirer/sdrtrunk/pull/1401
* #1981 P25P1 Foreign System IDEN_UPD_TDMA Message Support by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1982
* #1983 Revert Aliasing of Fully Qualified Identifiers Against Local ID by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1984
* #1915 (P25) Audio Stops Playing While Calls Continue To Process by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1986
* #1985 Update to JDK 23 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1987
* #1990 Ignores GPS positions with 0/0 latitude and longitude values. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1991
* #1994 P25 Phase1 Missing Subsequent Calls Audio When Multiple Conversants On Same Traffic Channel by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1995
* #1996 P25P1 Traffic Channel Monitor Clears Allocated Traffic Channels by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/1997
* #1914 P25 Talker Alias Support by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2001
* #2002 P25 Fully Qualified Identifier Formatting For Not Fixed Width Preference by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2003
* #2004 P25P1 Link Control Extended Source Message Processing & ISSI Radio IDs by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2005
* #2007 P25 Call events tracking.  P25 Traffic channel manager now uses… by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2010
* #2006 AudioOutput Recreated Once It Fails To Accept Audio Samples For Playback by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2011
* #2013 Enhance SUID extension message handling and correct encryption … by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2014
* #1973 JUnit Test for Patched Talkgroup Audio Call Streaming by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2015
* #2016 Playback Preferences Null Error by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2017
* #2018 P25 Phase 1 Now Playing Flickering Identifiers by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2019
* #2020 P25 channel null frequency band error. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2021
* #2022 Audio output reset now reacquires gain and mute controls. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2023
* #2025 P25P1 Audio Module Null Codec Error. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2027
* #2030 Playlist Editor Double-Click Channel Start by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2031
* #2032 Audio Output Logs Failed Attempt To Update Gain/Mute Controls by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2033
* #2035 P25P1 Harris Patched Talkgroup Values Incorrect by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2036
* #2028 Windows ARM/aarch64 Operating System Support by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2029
* #1060 Alias List Delete and Rename Operations To Playlist Editor @whelgeson by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2038
* #2039 P25 Phase 2 Incorrect Talkgroup & Radio Values In Call Metadata by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2042
* #2043 DMR Tier 3 Capacity Max Incorrect Talkgroup Values by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2044
* #2045 P25 Phase 1 Data Channels Always Ignored by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2046
* #2037 JavaFX Windows Can't Be Shrunken by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2047
* #2040 Reset AMBE/IMBE Audio Codec After Each Call by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2048
* #2034 Map Plotted Elements Timestamp Includes Date by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2049
* #1867 Event Logger - Log Radio ID and Not Network Radio Alias by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2050
* #2055 Multi-Frequency channel source with channel rotation monitor st… by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2056
* #1992 Traffic channels stuck in teardown. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2061
* #2062 Verbose logging option for Broadcastify Feed to troubleshoot st… by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2063
* #2064 Moto talker GPS event TO/FROM identifiers. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2065
* #2066 P25 Phase 1 Frequency Band Link Control Messages Checked For Valid CRC by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2067
* #2068 P25 traffic channel manager ignores traffic channel requests if… by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2069
* #2055 MultiFrequencyTunerChannelSource Null Pointer Error On Channel … by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2070
* #2071 Update To Version 0.6.1 Beta 3 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2072
* #2074 DMR CapMax LRRP GPS Alias List by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2075
* #2076 DMR CapMax Network and Site vales off by one. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2077
* #2078 Broadcastify Feed Verbose Logging Includes Audio and Metadata by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2080
* #2081 Windows aarch64 libusb support. by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2083
* #2084 Updates usb4java native library for windows aarch64 by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2085
* #2086 Audio Playback Manager Memory Leak by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2087
* #2089 Motorola P25P2 TDMA data channel support by @DSheirer in https://github.com/DSheirer/sdrtrunk/pull/2090
* Add Motorola OTA alias decoding by @ilyacodes in https://github.com/DSheirer/sdrtrunk/pull/2098

## New Contributors
* @poggs made their first contribution in https://github.com/DSheirer/sdrtrunk/pull/1813
* @ilyacodes made their first contribution in https://github.com/DSheirer/sdrtrunk/pull/2098

**Full Changelog**: https://github.com/DSheirer/sdrtrunk/compare/v0.6.0...v0.6.1

<hr /><em>This discussion was created from the release <a href='https://github.com/DSheirer/sdrtrunk/releases/tag/v0.6.1'>Version 0.6.1 (final)</a>.</em>

**Comment by funman1steve** [comment]
Just a heads up tried to download the 64bit Win and got this....
![image](https://github.com/user-attachments/assets/e238f55a-d5b4-4191-9e52-d606e8ea509d)

**Comment by MNSDR** [comment]
Download the other one it works great

**Comment by MNSDR** [comment]
I never got that and I got top of the line security

**Comment by funman1steve** [comment]
No clue, it might be a false positive from Microsoft??
Looks like many others are getting the AV tripping too on the FB post... :(

**Comment by ChiefGeekOfficer** [comment]
Ever since the 0.6.1 betas and now including the final, SDRTrunk doesn't seem to be able to decode most calls. I can see the calls in the status, but nothing plays. Downgrading to 0.6.0 works every time. Has something changed to prevent this from working? I am using Nooelec RTL-SDR V5 Dongles. Running on Linux.

UPDATE: Fixed by dialing in the tuners, seems new version is more sensitive to frequency drift.

**Comment by nasturtevant** [comment]
I'm following the Getting Started guide for **OS-X x86 64-bit**. It looks like the shell script is missing under **./bin/** instead there is a windows .bat file [sdr-trunk-osx-x86_64-vnightly.zip](https://github.com/DSheirer/sdrtrunk/releases/download/nightly/sdr-trunk-osx-x86_64-vnightly.zip)

**Comment by funman1steve** [comment]
Not sure on the Mac version, but for the linux version you run this one...
[image: image.png]
